File: scripts/validate-video-prompt.py
#!/usr/bin/env python3

import argparse
from pathlib import Path
import json
import sys

from video_prompt_validation import validate_video_prompt


def main():
    parser = argparse.ArgumentParser(description="Validate a video prompt against model capabilities and official vendor prompt structure.")
    source = parser.add_mutually_exclusive_group()
    source.add_argument("--prompt")
    source.add_argument("--prompt-file")
    parser.add_argument("--video-model", default="minimax-h3")
    parser.add_argument("--duration", type=float, default=10)
    parser.add_argument("--image-count", type=int, default=0)
    parser.add_argument("--video-count", type=int, default=0)
    parser.add_argument("--audio-count", type=int, default=0)
    parser.add_argument("--prompt-mode", choices=["T2VA", "I2VA", "FL2VA", "L2VA", "Ref2VA"])
    parser.add_argument("--prompt-source", default="user")
    args = parser.parse_args()
    prompt = args.prompt or (Path(args.prompt_file).read_text(encoding="utf-8") if args.prompt_file else sys.stdin.read())
    try:
        result = validate_video_prompt(prompt, args.video_model, args.duration, args.image_count, args.video_count, args.audio_count, args.prompt_mode, args.prompt_source)
    except ValueError as error:
        print(f"Error: {error}", file=sys.stderr)
        raise SystemExit(1)
    print(json.dumps(result, ensure_ascii=False, indent=2))


if __name__ == "__main__":
    main()

File: scripts/validate-video-prompt.mjs
#!/usr/bin/env node

import fs from "node:fs";
import { validateVideoPrompt } from "./video-prompt-validation.mjs";

function die(message) {
  console.error(`Error: ${message}`);
  process.exit(1);
}

function parseArgs(argv) {
  const args = {};
  const valued = new Set(["prompt", "prompt-file", "video-model", "duration", "image-count", "video-count", "audio-count", "prompt-mode", "prompt-source"]);
  for (let index = 0; index < argv.length; index += 1) {
    const token = argv[index];
    if (!token.startsWith("--")) die(`unexpected argument: ${token}`);
    const key = token.slice(2);
    if (!valued.has(key)) die(`unsupported option: --${key}`);
    const value = argv[index + 1];
    if (!value || value.startsWith("--")) die(`missing value for --${key}`);
    args[key] = value;
    index += 1;
  }
  return args;
}

const args = parseArgs(process.argv.slice(2));
const prompt = args.prompt ?? (args["prompt-file"] ? fs.readFileSync(args["prompt-file"], "utf8") : !process.stdin.isTTY ? fs.readFileSync(0, "utf8") : "");
try {
  const result = validateVideoPrompt({
    prompt,
    model: args["video-model"] || "minimax-h3",
    duration: Number(args.duration || 10),
    imageCount: Number(args["image-count"] || 0),
    videoCount: Number(args["video-count"] || 0),
    audioCount: Number(args["audio-count"] || 0),
    promptMode: args["prompt-mode"],
    promptSource: args["prompt-source"] || "user",
  });
  console.log(JSON.stringify(result, null, 2));
} catch (error) {
  die(error.message || String(error));
}
